File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int,input().split()) #[10, 500]#
card_numbers = list(map(int,input().split())) #[93, 181, 245, 214, 315, 36, 185, 138 ,216, 295] #


card_list= []
for fist in range(len(card_numbers)-2):
    for second in range(fist+1,len(card_numbers)-1): #마지막 수는 들어가지 않는다
        for third in range(second+1,len(card_numbers)):
            card_list.append([card_numbers[fist], card_numbers[second],card_numbers[third]])


similar_result = 0
for combination  in card_list:
    result = sum(combination)

    #목표값보다 작거나 같은 최대 합계를 찾는다
    if result <= m and result > similar_result:
        similar_result = result


print(similar_result)

File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.io.*;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String[] input = br.readLine().split(" ");
        String[] cards = br.readLine().split(" ");
        int  N = Integer.parseInt(input[0]);
        int  M = Integer.parseInt(input[1]);


        ArrayList<Integer> sumList = new ArrayList<>();
        for(int first = 0; first <N-2; first++){
            for(int second = first+1; second < N-1; second++){
                for(int third = second+1; third < N; third++){
                    sumList.add(Integer.parseInt(cards[first])+Integer.parseInt(cards[second])+Integer.parseInt(cards[third]));
                }
            }
        }
        int result = 0;
        for(int sum : sumList){
            if(sum <= M && result<= sum){
                result = sum;
            }
        }

        bw.write(String.valueOf(result));

        bw.flush();
        br.close();
        bw.close();
    }
}



